Issue 1: refactor codingstandards

sandrab opened the pull request.

sandrab committed:
Add composer, phpcs and editor config files

sandrab committed:
Fix main plugin file coding standards and readme

sandrab committed:
Refactor the code

Use psr-4 autoload to load PHP classes, improved directory structure for PHP classes.
Add @wordpress/scripts npm package to use for building javascript assets and checking coding standards.
Removed unused code and images.

sandrab committed:
Fix bug when custom frontend URL is empty

sandrab committed:
Add an Updater to migrate data when the plugin is updated

For example migrating option records when the plugin is updated from an older version.

sandrab committed:
Fix bug with alignment cancel-link in loginform

sandrab committed:
Rename nonce property passed from php to js

sandrab committed:
Rename one more nonce property

sandrab committed:
Improve login form to allow multiple forms on one screen

Use classes instead of id's in the html and do the manipulations/actions relative to the form the visitor acted upon.

sandrab committed:
Only localize the login script once

Otherwise the cyclosLoginObj will be defined multiple times in the HTML if the login form happens to be on the screen multiple times.

sandrab committed:
Use wp-i18n to make javascript strings translatable

sandrab committed:
Improve Updater and migrate old translations

Bugfix after Updater class was moved. Migrate old option records containing labels.
Remove old option records containing error messages, they can no longer be customized.

sandrab committed:
Remove cyclos_version option record on uninstall

sandrab committed:
Add notes about the channel in the requests

sandrab committed:
Restrict returned fields in auth request

sandrab committed:
Fix phpcs problem with % character in string

sandrab committed:
Fix bug with wrong references to moved classes

sandrab committed:
Fix bug in Admin settings when WP calls sanitize callback twice

sandrab committed:
Increase minimum WP core version to 5.0

Mainly because we use the new and recommended 5.0 way of translating JavaScript strings with wp-i18n.

sandrab committed:
Fix bug registering activation and uninstall hooks

sandrab committed:
Move uninstall hook into activation function

Otherwise the uninstall hook is registered again while uninstalling the plugin.
This would leave the uninstall information of the deleted plugin in the uninstall_plugin options record which is not nice.

sandrab committed:
Avoid php notice when connection fields are empty

sandrab committed:
Use wiki homepage until we have a dedicated wiki page

sandrab committed:
Improve explanation texts in admin screen

sandrab committed:
Keep original widget id for backwards compatibility

sandrab committed:
Remove widget data on plugin uninstall

sandrab committed:
Add new uninstall hook on update to 2.0.0

sandrab committed:
Improve check on minimum requirements

sandrab committed:
Refactor the Updater into a Setup class and improve

sandrab committed:
Increase PHP requirement to 7.2

Also make the code before we check the minimum PHP version work on older versions as well (tested PHP5.5).

sandrab committed:
Bugfix when Cyclos does not require captcha for forgot-password

sandrab committed:
Only load the Admin component on the WP admin

sandrab committed:
Use phpcs:ignore instead of :disable/enable if possible

sandrab committed:
Improve admin settings: split status from setup

sandrab committed:
Remove Admin render method for unused 'info' type

sandrab committed:
Replace composer autoload with custom autoload

sandrab committed:
Update @wordpress/scripts package

luisfpg reviewed (approved):
As mentioned before, my PHP / WordPress knowledge is minimal.
I've focused more in the API usage and haven't spotted any issues.
The only thing to take into account in a future issue of the plugin, is that Cyclos 4.13 has changed how the forgot password works, to allow using SMS. The old way will be removed in Cyclos 4.15.
The new way is that the user can receive by e-mail or SMS a code. So, instead of a link with a large token, the user just gets a 6-digit verification code, and enters it to change their password.
For reference, this is the issue in the new frontend (has a related commit): https://github.com/cyclosproject/cyclos4-ui/issues/290

sandrab commented:
Thanks Luis. Just created issue #3 for the forgot password retrieval.

sandrab committed:
Emphasize importance of a dedicated Cyclos user

sandrab committed:
Add convenience link to settings in Plugins list

sandrab committed:
Adjust the readme

Improved installation instructions, removed changelog (moved to wiki), removed upgrade notice (does not seem to work anymore in WP) and some minor changes.

sandrab committed:
Adjust link to wiki page
